pkg/storegateway/bucket_stores.go

```diff
@@ -428,6 +457,49 @@ func (u *BucketStores) getOrCreateStore(userID string) (*store.BucketStore, erro
 	return bs, nil
 }
 
+// deleteLocalFilesForExcludedTenants removes local "sync" directories for tenants that are not included in the current
+// shard.
+func (u *BucketStores) deleteLocalFilesForExcludedTenants(includeUserIDs map[string]struct{}) {
+	files, err := ioutil.ReadDir(u.cfg.BucketStore.SyncDir)
+	if err != nil {
+		return
+	}
+
+	for _, f := range files {
+		if !f.IsDir() {
+			continue
+		}
+
+		userID := f.Name()
+		if _, included := includeUserIDs[userID]; included {
+			// Preserve directory for users owned by this shard.
+			continue
+		}
+
+		removed, bs := u.removeEmptyBucketStore(userID)
+		if bs != nil {
+			if !removed {
+				continue
+			}
+
+			err := bs.Close()
+			if err == nil {
+				level.Info(u.logger).Log("msg", "closed bucket store for user", "user", userID)
+			} else {
+				level.Info(u.logger).Log("msg", "failed to close bucket store for user", "user", userID, "err", err)
```

Comment:
Is it an info? I think warn level may be more appropriate here. All in all, it's an unexpected error condition.